### ISSUE TITLE
Multiple account sign in and switching

### ISSUE THREAD

**Issue by Thaina** [microsoft/vscode#166176]
Sometimes it was more convenient to have multiple account; such as, organization account, administration account, company provided account, separate from owned individual account and can manage each project separately or having separate loaded extension and setting profile

For example I myself have my own current account that mainly write C# and unity. But I also have responsibility on devops and server in my company, which mainly use js and some specific tools and access to company repository in full access mode, which I don't want to do that with my own account and I don't want to (openly) add my account to be collaborator of those projects or organization

Also sometimes some company will provide account for employee for managerial purpose

So I wish that I could login with multiple account in vscode without signing out. And when I open an instance of vscode project/folder/workspace I can switch account, and make it become default of that project so when reopen it will auto select account that fit for the project for each instance

**Comment by TylerLeonhardt** [comment]
@sandy081 this sounds like Profiles

**Comment by sandy081** [comment]
Can you please try our latest preview feature - Profiles (https://code.visualstudio.com/updates/v1_69#_settings-profiles) and let us know if it suffice your needs.

**Comment by Thaina** [comment]
@sandy081 I have tried that new setting profile feature. I have expect that I can switch github account by profile. I can't find a way to do that so I have create this issue

**Comment by sandy081** [comment]
I see. Currently profiles are accounts are separate concepts. Environment switching only happens through profiles. Have you tried creating different profiles for your use cases and log in to different accounts in each profile? The only difference you have to make is to use profiles to switch but not accounts. Switching a profile can also switch the account you logged in.

CC @isidorn

**Comment by Thaina** [comment]
@sandy081 Thanks you for your suggestion. I will try that

**Comment by isidorn** [comment]
@sandy081 thanks for the ping. Let's see if more users find this confusing. I will also make sure to ask users in the user study.

**Comment by Thaina** [comment]
I have tried creating new setting profile. Then logout and login second github account. When I switch back to default profile the second account still persist

When I try to sign out and sign in first account in default profile, then switch profile to second profile, it still is first account

I am understood that account and profile are separate concept. So there is 2 way in my mind. One is to have each of these feature can be switching in each project or workspace independently. Second option is to have profile depend on account and so setting sync will sync profile into account separately, and just allow copy or clone profile from another account

**Comment by sandy081** [comment]
@TylerLeonhardt I remember we discussed that whether we wanna make accounts profile specific? I think this is one such use case where user wants this behaviour.

**Comment by isidorn** [comment]
@sandy081 I strongly feel that accounts should be global, and not stored in a profile.
Not sure what is the exact use case where you would want account to be a part of a profile. I also think we should strive towards - users just login once and all services light up.

**Comment by TylerLeonhardt** [comment]
Yes we don't want accounts to be tied to specific profiles exactly as @isidorn has summarized.

I understand that folks might want to be able to log in to multiple GitHub accounts, but that's a totally separate issue that has nothing to do with profiles: https://github.com/microsoft/vscode/issues/127967

**Comment by VSCodeTriageBot** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!

**Comment by Thaina** [comment]
@TylerLeonhardt It might be out of scope of setting profiles, but is it out of scope of vscode too?

**Comment by TylerLeonhardt** [comment]
@Thaina switching GitHub accounts within VS Code is tracked in the issue I linked. We want to do it, but are blocked by GitHub's current experience.

**Comment by Thaina** [comment]
@TylerLeonhardt I have made this issue because I think about alternative approach, the active account always be one account. Just that vscode remember a list of token and allow switch account per instance (and remember the default in each project)

This would not solve the copilot collision though

**Comment by TylerLeonhardt** [comment]
I think there are other signals to tell us what account to use.

For settings, extensions, etc we have Profiles.

For something like the GitHub Pull Request extension, that uses the git origin (and other things) to determine if the repo is a GitHub repo or a GitHub Enterprise Server repo and asks for the auth of the correct one.

Since we don't have multiple GitHub account support, it's hard to design what that will look like when GitHub finally adds support on GitHub.con for multi accounts

**Comment by Thaina** [comment]
@TylerLeonhardt I don't think it would require github to support multiple account and I don't think github would do that. Just that vscode support multiple sign in account stored in vscode. It did not meant to be active and could use action of every account at once. As I said it still only one active

What I imagine is when we sign in to github, vscode just got the oauth token to use in some actions. The key is vscode could store all token and have UI for switching account. In effect it like we sign out and sign in to another account, but skip all the flow

In essence it's similar to google account list, or the chrome profile. And just also remember the last account of the project

And maybe in the future we could allow each extension or command or task to select or specify account for its action

**Comment by Pablo-Oliveira** [comment]
I am particularly interested in the option of multiple Microsoft account logins to be able to manage multiple remote access tunnels.
I personally work on a single project at a time, but very often I need to provide support in multiple production environments.
Furthermore, using two-factor authentication gives me a lot of security and allows me to free myself from the need to manage multiple ssh keys.

**Comment by Thaina** [comment]
In the end it seem it is not out of scope

https://code.visualstudio.com/updates/v1_95#_multiple-github-accounts

This issue should not be closed prematurely

**Comment by TylerLeonhardt** [comment]
We support multiple GitHub accounts but the intent is that we don't want profiles to be 1-to-1 with accounts. Which is why this was marked as out of scope. You can sign in to multiple accounts, but those accounts are shared across profiles.

**Comment by Thaina** [comment]
@TylerLeonhardt I have already explicitly stated in https://github.com/microsoft/vscode/issues/166176#issuecomment-1323720504 that I create this issue because what I want is not relate to profile and my use case cannot be solved from profile from the start

**Comment by TylerLeonhardt** [comment]
I see @Thaina. Then, https://code.visualstudio.com/updates/v1_95#_multiple-github-accounts solves this for you since you are able to switch the account you use with the `GitHub` extension.
